client/src/documentLinkProvider.ts

```diff
@@ -27,20 +27,49 @@ export class BitbakeDocumentLinkProvider implements vscode.DocumentLinkProvider
-    const pnDir = path.join(parentDir, extractRecipeName(document.uri.path) as string)
+    const parentDir = uri.path.split('/').slice(0, -1).join('/')
+    const pnDir = path.join(parentDir, extractRecipeName(uri.path) as string)
     const filesDir = path.join(parentDir, 'files')
-    return [...(await vscode.workspace.findFiles(new vscode.RelativePattern(pnDir, '**/' + filenamesRegex), undefined, filenames.length, token)),
-      ...(await vscode.workspace.findFiles(new vscode.RelativePattern(filesDir, '**/' + filenamesRegex), undefined, filenames.length, token))]
+    try {
+      return [...(await vscode.workspace.findFiles(new vscode.RelativePattern(pnDir, '**/' + filenamesRegex), undefined, filenames.length, token)),
+        ...(await vscode.workspace.findFiles(new vscode.RelativePattern(filesDir, '**/' + filenamesRegex), undefined, filenames.length, token))]
+    } catch (err) {
+      logger.error(`An error occurred when finding files with pattern: ${filenamesRegex}. ${JSON.stringify(err)}`)
+      return []
+    }
+  }
+
+  private findFolders (links: Awaited<RequestResult['getLinksInDocument']>): Awaited<RequestResult['getLinksInDocument']> {
+    return links.filter(link => path.parse(link.value).ext === '')
   }
 
   async provideDocumentLinks (document: vscode.TextDocument, token: vscode.CancellationToken): Promise<vscode.DocumentLink[]> {
     const linksData = await this.client.sendRequest<RequestResult['getLinksInDocument']>(RequestMethod.getLinksInDocument, { documentUri: document.uri.toString() })
     const documentLinks: vscode.DocumentLink[] = []
-    const relatedFiles = await this.findRelatedFiles(document, linksData, token)
-    logger.debug(`Found ${relatedFiles.length} local SRC_URI files for ${document.uri.toString()}`)
+    const relatedFiles = await this.findRelatedFiles(document.uri, linksData, token)
+    const folders = this.findFolders(linksData)
+
+    logger.debug(`Found ${folders.length} folders from the links for file: ${document.uri.toString()}`)
+    logger.debug(`Found ${relatedFiles.length} files from the links for file: ${document.uri.toString()}`)
+
+    // Reveal folders in explorer
+    folders.forEach((folder) => {
+      const parentDir = document.uri.path.split('/').slice(0, -1).join('/')
+      const pnDir = path.join(parentDir, extractRecipeName(document.uri.path) as string)
+      const commandArguments = [path.join(pnDir, folder.value)]
+      const targetUri = vscode.Uri.parse(`command:bitbake.revealInExplorer?${encodeURIComponent(JSON.stringify(commandArguments))}`)
+      /*
+        targetUri.path === targetUri.fsPath === "bitbake.revealInExplorer",
+        targetUri.scheme === "command",
+        targetUri.query is the array of arguments
+
+        Assigning command uris to document links will trigger the command on ctrl+click with the arguments
+      */
+      documentLinks.push(new vscode.DocumentLink(folder.range, targetUri))
+    })
```

Comment:
It looks like links to file got broken again. They try to open at the root of the file system instead. For instance `file://mdev.conf`

Reply:
Somehow both our link provider and the default provider in vscode return something. And `ctrl+click` selects the result from the default one.
![image](https://github.com/yoctoproject/vscode-bitbake/assets/47988425/824ff5e5-de54-4556-83cd-fa2be44fd4c4)

Doesn't seem there is a way to control the behaviour of the default provider and we probably don't want to.

Reply:
This didn't happen previously, at least, our link was more prioritized. This is an important regression compared to the previous version. The current PR breaks most links to provide a few directories link. We need to fix this or abandon the change.

Reply:
check 5849c1e53cbfc5173e7912dd88c56a1bf3981c20, I didn't do anything specifically to fix this but this commit should have better results.

Reply:
Now I get back the file links, but directory 'reveal' links no longer exist on my end. I basically go back to the previous behavior.

Reply:
The link from the default provider is probably taking priority before the command URL is resolved. On my end, the correct command uri will be ready after some delay since I made it `async`. 
![image](https://github.com/yoctoproject/vscode-bitbake/assets/47988425/e5a1d77a-7d37-4070-a360-8f2f50ac1f08)
The 'ctrl+click' seems always to point to the top most option. So in this case, the 'reveal' should be triggered.

Reply:
sorted the values before creating the regex, this should make sure all the intended dirs will be returned.
Also added a [line](https://github.com/yoctoproject/vscode-bitbake/pull/82/commits/b4697862843f8226ef0df3e797187bba71ab269f#diff-a37941234785530674e097388962bd0931e8450c872a79fd3d3a163ba9b528eaR94) with find.dirSync(), you may comment out the async one and run with the sync one to see if the priority issue is better handled. I am actually out of ideas.

Reply:
Your latest iteration is much better! we only get duplicate links for directories now. [this discussion](https://stackoverflow.com/questions/74537129/how-can-i-make-clickable-file-position-links-in-an-output-channel-that-open-the) explains how the default link provider works. We succesfully get rid of the original links for files, I wanna try to understand why we can't do that for dirs.

Reply:
_union in vscode seem to do the duplication removal. With the debugger, the ranges seem exactly identical. async seem to change things, we get inversed results. If I remove all but 2 links, it works with sync. We must provide links in the exact same order to unionize properly???
That was it. Fixed and pushed :rocket:

Reply:
Fantastic!